psydac/fem/partitioning.py

```diff
@@ -52,11 +52,12 @@ def partition_coefficients(domain_decomposition, spaces, min_blocks=None):
 
     if min_blocks is None:
         min_blocks = [None]*ndims
+
     for s,e,V,mb in zip(global_starts, global_ends, spaces, min_blocks):
         if mb is None:
-            assert all(e-s+1>=V.degree*(1-V.periodic)+V.periodic)
+            assert all(e-s+1>=V.degree)
         else:
-            assert all(e-s+1>=mb*(1-V.periodic)+V.periodic)
+            assert all(e-s+1>=mb*(1-V.periodic)+V.degree*V.periodic)
```

Comment:
Here we assume that each process has at least `degree` coeffs, before we didn't have this condition in the periodic case.

Reply:
in the periodic case, we don't need `mb` only in the non-periodic case where we have this option.
this option is needed for the multipatch case, where we need `V.degree+1` instead of `V.degree `, which can only happen in the non-periodic case

Reply:
All right, so you want to ignore the value of `mb` in the periodic case, and use it only in the non-periodic case (if it is not `None`). Why not writing this then?

```python
if V.periodic or mb is None:
    assert all(e - s + 1 >= V.degree)
else:
    assert all(e - s + 1 >= mb)
```

Reply:
okay. I will make the changes

Reply:
done in [78e0f2e](https://github.com/pyccel/psydac/pull/240/commits/78e0f2e9fcb32efef034ccb24ec2859d68cb200d)